File: Python/slash_commands/flush_telegram.py
"""/flush-telegram - force flush of Telegram queue (admin only)."""
from logger import Logger
from telegram_bridge import get_bridge
logger = Logger()
name = 'flush-telegram'
description = "Force l'envoi immédiat des messages en file vers Telegram (Administrateur uniquement)"

async def execute(interaction, **kwargs):
    try:
        # basic admin check
        member = getattr(interaction, 'member', None)
        is_admin = False
        try:
            is_admin = getattr(member.guild_permissions, 'administrator', False)
        except Exception:
            is_admin = False
        if not is_admin:
            await interaction.response.send_message('Vous devez être administrateur pour utiliser cette commande.', ephemeral=True)
            return

        await interaction.response.defer(ephemeral=True)
        tg = get_bridge()
        try:
            # call internal flush implementation if available
            if hasattr(tg, '_flush'):
                tg._flush()
                await interaction.followup.send("Flush Telegram exécuté (tentative d'envoi immédiat).", ephemeral=True)
            else:
                await interaction.followup.send("Le module Telegram n'est pas disponible sur ce serveur.", ephemeral=True)
        except Exception as e:
            await interaction.followup.send(f'Erreur lors du flush Telegram: {e}', ephemeral=True)
    except Exception as err:
        logger.error(['Erreur /flush-telegram:', err])
        try:
            await interaction.followup.send('Erreur interne.', ephemeral=True)
        except Exception:
            pass

File: V2/src/slash-commands/flush-telegram.js
import { SlashCommandBuilder, PermissionsBitField } from 'discord.js';
import logger from '../logger.js';

export default {
  data: new SlashCommandBuilder()
    .setName('flush-telegram')
    .setDescription('Force l\'envoi immédiat des messages en file vers Telegram (Administrateur uniquement)'),
  async execute(interaction) {
    try {
      const member = interaction.member;
      const isAdmin = member && member.permissions && member.permissions.has(PermissionsBitField.Flags.Administrator);
      if (!isAdmin) {
        await interaction.reply({ content: 'Vous devez être administrateur pour utiliser cette commande.', ephemeral: true });
        return;
      }

      await interaction.deferReply({ ephemeral: true });
  const { default: telegram } = await import('../telegram.js');
      if (!telegram || typeof telegram._flush !== 'function') {
        await interaction.editReply({ content: 'Le module Telegram n\'est pas disponible sur ce serveur.' });
        return;
      }

      try {
        await telegram._flush();
        await interaction.editReply({ content: 'Flush Telegram exécuté (tentative d\'envoi immédiat).' });
      } catch (e) {
        await interaction.editReply({ content: 'Erreur lors du flush Telegram: ' + (e && e.message ? e.message : String(e)) });
      }
    } catch (err) {
      try { logger.error(['Erreur /flush-telegram:', err]); } catch (e) {}
      if (interaction.replied || interaction.deferred) await interaction.followUp({ content: 'Erreur interne.', ephemeral: true }); else await interaction.reply({ content: 'Erreur interne.', ephemeral: true });
    }
  }
};
